esx_service/cli/vmdkops_admin.py

```diff
@@ -726,38 +731,382 @@ def get_version():
+    
+    if not tenant:
+        error_info = "Tenant {0} does not exist".format(args.name)
+        return operation_fail(error_info)
+
+    error_info, vms = generate_tuple_from_vm_list(args.vm_list)
+    
+    if error_info:
+        return operation_fail(error_info)
+      
+    error_info = tenant.remove_vms(auth._auth_mgr.conn, vms)
+
+    if error_info:
+        return operation_fail(error_info)
+    else:
+        print "tenant vm rm succeeded"
+
+def tenant_vm_ls_headers():
+    """ Return column names for tenant vm ls command """
+    headers = ['Uuid', 'Name']
+    return headers
+
+def generate_tenant_vm_ls_rows(vms):
+    """ Generate output for tenant vm ls command """
+    rows = []
+    for vm in vms:
+        # vm has the format like this (vm_uuid, vm_name)
+        uuid = vm[0]
+        name = vm[1]
+        rows.append([uuid, name])
+    
+    return rows
+
 
 def tenant_vm_ls(args):
-    print(NOT_IMPLEMENTED)
+    """ Handle tenant vm ls command """
+    error_info, tenant = get_tenant_from_db(args.name)
+    if error_info:
+        return operation_fail(error_info)
+    
+    if not tenant:
+        error_info = "Tenant {0} does not exist".format(args.name)
+        return operation_fail(error_info)
+   
+    header = tenant_vm_ls_headers()
+    rows = generate_tenant_vm_ls_rows(tenant.vms)
+    print(cli_table.create(header, rows))
+
+def default_privileges():
+     privileges = [{'datastore': 'datastore1',
+                     'global_visibility': 0,
+                      'create_volume': 0,
+                      'delete_volume': 0,
+                     'mount_volume': 0,
+                      'max_volume_size': 0,
+                      'usage_quota': 0}]
+     return privileges
+
+def set_privileges(rights, privileges, value):
+    """ set or unset privileges based rights passed by command line """
+    if 'create' in rights:
+        privileges[auth_data_const.COL_CREATE_VOLUME] = value
+
+    if 'delete' in rights:
+        privileges[auth_data_const.COL_DELETE_VOLUME] = value
+
+    if 'mount' in rights:
+        privileges[auth_data_const.COL_MOUNT_VOLUME] = value
+
+    if 'all' in rights:
+        privileges[auth_data_const.COL_CREATE_VOLUME] = value
+        privileges[auth_data_const.COL_DELETE_VOLUME] = value
+        privileges[auth_data_const.COL_MOUNT_VOLUME] = value
+
+def generate_privileges(args):
+    """ Generate privileges based on CLI argument """
+    privileges = default_privileges()[0]
+    privileges[auth_data_const.COL_DATASTORE] = args.datastore
+    
+    if args.rights:
+        set_privileges(args.rights, privileges, 1)
+    
+    if args.volume_maxsize:
+        size_in_MB = convert.convert_to_MB(args.volume_maxsize)
+        privileges[auth_data_const.COL_MAX_VOLUME_SIZE] = size_in_MB
+    
+    if args.volume_totalsize:
+        size_in_MB = convert.convert_to_MB(args.volume_totalsize)
+        privileges[auth_data_const.COL_USAGE_QUOTA] = size_in_MB
```

Comment:
There should be validation that the volume max size and max capacity across volumes isn't specified beyond the capacity of the datastore itself. Or larger or smaller than VSphere limits. Possibly any value larger than the datastore capacity is simply capped at the datastore capacity.